crates/ironrdp-pdu/src/geometry.rs

```diff
@@ -22,11 +22,11 @@ impl Rectangle {
     }
 
     pub fn width(&self) -> u16 {
-        self.right - self.left + 1
+        self.right - self.left
     }
 
     pub fn height(&self) -> u16 {
-        self.bottom - self.top + 1
+        self.bottom - self.top
     }
```

Comment:
Did you try running the web client? IIRC, `width()` and `height()` are used when sending partial image updates, and this change may break that.

Also, my understanding is that an update rectangle such that: `left = 0`, `top = 0`, `right = 63` and `bottom = 63`, means we have a chunk of 64x64 pixels, hence the `width()` and `height()` should probably return 64, not 63.
I suspect the actual problem lies in the consumer code, the rfx module itself. I would check where `width()` and `height()` are called and see if there is occurrence of such code:
```rust
let some_rectangle = …;
let a_reconstructed_rectangle = Rectangle {
    left: some_rectangle.left,
    top: some_rectangle.top,
    right: some_rectangle.left + some_rectangle.width(), // <-- bad
    bottom: some_rectangle.top + some_rectangle.height(), // <-- bad
};
```
Seeing the change you made in the `clipping_rectangles`, that’s likely what’s happening somewhere.

Reply:
I had to fix failing unit tests as a result of my change, but I'd rather standardize our definition of rectangles to the one where right == width, bottom == height, and then do the +1, -1 when serializing/deserializing from packets that use a different definition. it appears to be working fine for me with and without RemoteFX, no black lines

Reply:
> I had to fix failing unit tests as a result of my change, but I'd rather standardize our definition of rectangles to the one where right == width, bottom == height, and then do the +1, -1 when serializing/deserializing from packets that use a different definition.

Sound good then.

> It appears to be working fine for me with and without RemoteFX, no black lines

Sorry for being persistent, but that’s what you get when running the web client? Can you make sure you’re not running on an old build by removing the `crates/ironrdp-web/pkg` folder and building again from scratch? You’ll not get black lines but rather visual artifacts (one row and line missing from the graphical update).

Reply:
Also, you need a _small_ graphical update: https://github.com/Devolutions/IronRDP/blob/8857c3c25ea40bd2a2f4d850f67b16b2306393c9/crates/ironrdp-web/src/image.rs#L19

Reply:
https://github.com/Devolutions/IronRDP/blob/8857c3c25ea40bd2a2f4d850f67b16b2306393c9/crates/ironrdp-web/src/image.rs#L78

https://github.com/Devolutions/IronRDP/blob/8857c3c25ea40bd2a2f4d850f67b16b2306393c9/crates/ironrdp-web/src/image.rs#L34

https://github.com/Devolutions/IronRDP/blob/8857c3c25ea40bd2a2f4d850f67b16b2306393c9/crates/ironrdp-web/src/image.rs#L35

Reply:
@CBenoit I only test with the command-line client, I don't have the web client set up for development

Reply:
Okay, no problem I’ll handle that.